src/Server/HTTPHandler.cpp

```diff
@@ -727,6 +734,26 @@ void HTTPHandler::processQuery(
         return false;
     };
 
+    auto role_params_it = params.find("role");
+    if (role_params_it != params.end())
+    {
+        std::vector<UUID> roles_ids;
+        const auto & access_control = context->getAccessControl();
+        const auto & user = context->getUser();
+        for (; role_params_it != params.end(); role_params_it++)
+        {
+            if (role_params_it->first == "role")
```

Comment:
@vitlibar, this is coming from Poco, namely, `Poco::ListMap`, if I am not mistaken; is it OK to add it there? I assume it is since it is included in the CH source code and not as a contrib module.

Reply:
I added a new `NameValueCollection::getAll` method to extract all the parameters. It might be useful in the future.

Reply:
Well, I thought about modifying `DB::HTMLForm` actually, but probably modifying `NameValueCollection` is also ok.